tests/test_replica.py

```diff
@@ -307,12 +307,14 @@ def test_touch_replicas(self, rse_factory, mock_scope, root_account):
 
         assert get_replica_atime({'scope': files1[0]['scope'], 'name': files1[0]['name'], 'rse_id': rse_id}) is None
         assert get_did_atime(scope=mock_scope, name=files1[0]['name']) is None
+        assert get_did_access_cnt(scope=mock_scope, name=files1[0]['name']) == 0
```

Comment:
```suggestion
        assert get_did_access_cnt(scope=mock_scope, name=files1[0]['name']) is None
```

Currently this test is failing:
```
 _____________________ TestReplicaCore.test_touch_replicas ______________________
tests/test_replica.py:310: in test_touch_replicas
    assert get_did_access_cnt(scope=mock_scope, name=files1[0]['name']) == 0
E   AssertionError: assert None == 0
E    +  where None = get_did_access_cnt(scope=mock, name='file_fb6e29bb3bb44585931fd624261e3e9f')
```

My understanding is that the DID is not initialized with `access_cnt = 0`, so `access_cnt` would be `None`. 
This seems to be the case for the `accessed_at` times for the DID and replica, as seen from the assertions on line 308-309.

The `values(accessed_at=accessed_at, access_cnt=models.DataIdentifier.access_cnt + 1)` could potentially fail with `TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'`. You should change that logic in `touch_replica` to match the one in `touch_dids`:

https://github.com/rucio/rucio/blob/b6854f44e1e3bf003497256aa0a1c6dc3e00628c/lib/rucio/core/did.py#L2865-L2866

But I'm also having doubts on whether it's a good idea to increment the `access_cnt` on the DID as part of `touch_replica` - maybe the DID `access_cnt` should only be incremented by `touch_dids`, and a separate replica `access_cnt` field should be added to the `RSEFileAssociation` and `CollectionReplica` tables, considering that they already have an `accessed_at` field. @bari12 thoughts?